packages/ui/src/2_molecules/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FC } from 'react';
+
+import classNames from 'classnames';
+
+import { IconNames } from '../../1_atoms';
+import { applyDataAttr } from '../../utils';
+import styles from './Pagination.module.css';
+import { PaginationButton } from './components/PaginationButton/PaginationButton';
+import { usePaginate } from './hooks/usePaginate';
+
+type PaginationProps = {
+  className?: string;
+  dataLayoutId?: string;
+  pageSize?: number;
+  page: number;
+  setPage: (page: number) => void;
```

Comment:
As mentioned earlier, this is not correct but I think that you can rename this to `onChange` and it should be called every time we have a page change (and it should be optional ofc).